src/clean-config.js

```diff
@@ -14,66 +14,72 @@ const SECTION_KEYS = {
   * @returns {PhishingDetectorConfiguration} Cleaned config
   */
 const cleanConfig = (config, listName) => {
-  const section = SECTION_KEYS[listName];
-  const newConfig = {
-    version: config.version,
-    tolerance: listName === 'blocklist' ? 0 : config.tolerance, // disable fuzzychecking for performance
-    fuzzylist: [...config.fuzzylist],
-    [SECTION_KEYS.allowlist] : [...config[SECTION_KEYS.allowlist]],
-    [SECTION_KEYS.blocklist] : [...config[SECTION_KEYS.blocklist]],
-  };
+  switch (listName) {
+    case 'allowlist':
+      return cleanAllowlist(config);
+    case 'blocklist':
+      return cleanBlocklist(config);
+  }
+};
 
-  const finalEntries = new Set();
-  const excludedEntries = new Set();
-  const detector = new PhishingDetector(newConfig);
-  const baseList = detector.configs[0][listName];
+const cleanAllowlist = config => {
+  // when cleaning the allowlist, we want to:
+  // - remove any domains which are subdomains of entries in the blocklist
+  // - remove any domains which do not need to be excluded from the fuzzylist
+
+  const fuzzyDetector = new PhishingDetector({
+    ...config,
+    [SECTION_KEYS['blocklist']]: [],
+    [SECTION_KEYS['allowlist']]: [],
+  });
+
+  const blocklistSet = new Set(config[SECTION_KEYS['blocklist']]);
+  const allowlistSet = new Set(config[SECTION_KEYS['allowlist']]);
+
+  const newAllowlist = Array.from(allowlistSet).filter(domain => {
+    const parts = domain.split(".");
+    for (let i = 1; i < parts.length - 1; i++) {
+      if (blocklistSet.has(parts.slice(i).join("."))) {
+        return true;
```

Comment:
Shouldn't this be `false`? At least if we want to do what the comment above suggests:

>remove any domains which are subdomains of entries in the blocklist

Reply:
Oops, that comment is incorrect - we want to do what the code says because otherwise there's no point in having the explicit allowlist entry